### ISSUE TITLE
Exception in Laravel Blade files

### ISSUE THREAD

**Issue by maicol07** [klesun/deep-js-completion#18]
Error in Laravel Blade files:

```
java.lang.NoSuchMethodError: 'com.intellij.lang.javascript.psi.JSType com.intellij.lang.javascript.psi.JSTypeEvaluationResult.getType()'
at org.klesun.deep_js_completion.contexts.SearchCtx.$anonfun$getWsType$5(SearchCtx.scala:59)
at scala.Option.flatMap(Option.scala:283)
at org.klesun.deep_js_completion.contexts.SearchCtx.getWsType(SearchCtx.scala:59)
at org.klesun.deep_js_completion.contexts.SearchCtx.findExprType(SearchCtx.scala:198)
at org.klesun.deep_js_completion.contexts.ExprCtx.$anonfun$findExprType$1(ExprCtx.scala:54)
at org.klesun.deep_js_completion.contexts.ExprCtx.$anonfun$findExprType$1$adapted(ExprCtx.scala:53)
at scala.collection.Iterator$$anon$10.nextCur(Iterator.scala:585)
at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:599)
at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:592)
at org.klesun.lang.DeepJsLang$It.hasNext(DeepJsLang.scala:284)
at org.klesun.deep_js_completion.completion_providers.PropNamePvdr$.org$klesun$deep_js_completion$completion_providers$PropNamePvdr$$resolveMems(PropNamePvdr.scala:141)
at org.klesun.deep_js_completion.completion_providers.PropNamePvdr.addCompletions(PropNamePvdr.scala:269)
at com.intellij.codeInsight.completion.CompletionProvider.addCompletionVariants(CompletionProvider.java:32)
at com.intellij.codeInsight.completion.CompletionContributor.fillCompletionVariants(CompletionContributor.java:155)
at com.intellij.codeInsight.completion.CompletionService.getVariantsFromContributors(CompletionService.java:78)
at com.intellij.codeInsight.completion.CompletionResultSet.runRemainingContributors(CompletionResultSet.java:154)
at com.intellij.codeInsight.completion.CompletionResultSet.runRemainingContributors(CompletionResultSet.java:146)
at com.intellij.codeInsight.completion.CompletionResultSet.runRemainingContributors(CompletionResultSet.java:142)
at com.intellij.codeInsight.template.impl.LiveTemplateCompletionContributor$1.addCompletions(LiveTemplateCompletionContributor.java:89)
at com.intellij.codeInsight.completion.CompletionProvider.addCompletionVariants(CompletionProvider.java:32)
at com.intellij.codeInsight.completion.CompletionContributor.fillCompletionVariants(CompletionContributor.java:155)
at com.intellij.codeInsight.completion.CompletionService.getVariantsFromContributors(CompletionService.java:78)
at com.intellij.codeInsight.completion.CompletionService.getVariantsFromContributors(CompletionService.java:61)
at com.intellij.codeInsight.completion.CompletionService.performCompletion(CompletionService.java:134)
at com.intellij.codeInsight.completion.BaseCompletionService.performCompletion(BaseCompletionService.java:41)
at com.intellij.codeInsight.completion.CompletionProgressIndicator.lambda$calculateItems$12(CompletionProgressIndicator.java:862)
at com.intellij.util.indexing.FileBasedIndex.lambda$ignoreDumbMode$0(FileBasedIndex.java:162)
at com.intellij.openapi.util.RecursionManager$1.computePreventingRecursion(RecursionManager.java:111)
at com.intellij.util.indexing.FileBasedIndexEx.ignoreDumbMode(FileBasedIndexEx.java:574)
at com.intellij.util.indexing.FileBasedIndex.ignoreDumbMode(FileBasedIndex.java:161)
at com.intellij.util.indexing.DumbModeAccessType.ignoreDumbMode(DumbModeAccessType.java:43)
at com.intellij.codeInsight.completion.CompletionProgressIndicator.calculateItems(CompletionProgressIndicator.java:858)
at com.intellij.codeInsight.completion.CompletionProgressIndicator.runContributors(CompletionProgressIndicator.java:846)
at com.intellij.codeInsight.completion.CodeCompletionHandlerBase.lambda$startContributorThread$6(CodeCompletionHandlerBase.java:353)
at com.intellij.codeInsight.completion.AsyncCompletion.lambda$tryReadOrCancel$5(CompletionThreading.java:172)
at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1097)
at com.intellij.codeInsight.completion.AsyncCompletion.tryReadOrCancel(CompletionThreading.java:170)
at com.intellij.codeInsight.completion.CodeCompletionHandlerBase.lambda$startContributorThread$7(CodeCompletionHandlerBase.java:345)
at com.intellij.codeInsight.completion.AsyncCompletion.lambda$startThread$0(CompletionThreading.java:95)
at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$2(CoreProgressManager.java:178)
at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:688)
at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:634)
at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:63)
at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:165)
at com.intellij.codeInsight.completion.AsyncCompletion.lambda$startThread$1(CompletionThreading.java:91)
at com.intellij.util.RunnableCallable.call(RunnableCallable.java:20)
at com.intellij.util.RunnableCallable.call(RunnableCallable.java:11)
at com.intellij.openapi.application.impl.ApplicationImpl$1.call(ApplicationImpl.java:265)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:668)
at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:665)
at java.base/java.security.AccessController.doPrivileged(Native Method)
at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:665)
at java.base/java.lang.Thread.run(Thread.java:834)
```

IDE infos:

```
PhpStorm 2021.1 EAP
Build #PS-211.5787.18, built on February 11, 2021
PhpStorm EAP User
Expiration date: March 13, 2021
Runtime version: 11.0.10+9-b1319.6 amd64
VM: Dynamic Code Evolution 64-Bit Server VM by JetBrains s.r.o.
Windows 10 10.0
GC: G1 Young Generation, G1 Old Generation
Memory: 2022M
Cores: 16
Registry: run.processes.with.pty=TRUE, ide.balloon.shadow.size=0
Non-Bundled Plugins: Batch Scripts Support (1.0.12), String Manipulation (8.10.191.000.0), com.chrisrm.idea.MaterialThemeUI (6.0.0), com.github.patou.gitmoji (1.7.0), com.intellij.ideolog (203.0.26.0), com.mallowigi (36.0), com.vexus2.intellij.openingithub (1.0.6), io.github.biezhi.plugins.gitmoji-unicode (1.0.5), zielu.gittoolbox (203.4.8), com.intellij.plugins.html.instantEditing (211.5787.4), com.funivan.idea.phpClean (2020.12.22), com.github.shiraji.yaemoji (1.0.7), de.espend.idea.laravel (0.15.4), com.kalessil.phpStorm.phpInspectionsEA (4.0.6), de.espend.idea.php.annotation (8.0.0), net.king2500.plugins.PhpAdvancedAutoComplete (1.1.0), lv.midiana.misc.phpstorm-plugins.deep-keys (2021.01.05.001), lv.midiana.misc.idea-plugins.deep-js-completion (2020.11.30.001), ru.adelf.idea.dotenv (2021.1)
```

**Comment by klesun** [comment]
Hi, thanks for the report.
Yeah, there is a compatibility break in `2021.1 EAP`.
Will try to get to it as soon as I can.

**Comment by klesun** [comment]
You probably want to remove the plugin until the new release with the fix for EAP.

**Comment by klesun** [comment]
Released fix in [2021.04.21.001](https://plugins.jetbrains.com/plugin/11478-deep-js-completion/versions/stable/117651)

It should become public in about two days - when JetBrains is done validating
